pkg/asset/installconfig/aws/aws.go

```diff
@@ -133,37 +132,46 @@ func getCredentials() error {
 		return err
 	}
 
-	tmpl, err := template.New("aws-credentials").Parse(`# Created by openshift-install
-# https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html
-[{{.Profile}}]
-aws_access_key_id={{.KeyID}}
-aws_secret_access_key={{.SecretKey}}
-`)
-	if err != nil {
-		return err
-	}
-
 	path := defaults.SharedCredentialsFilename()
 	logrus.Infof("Writing AWS credentials to %q (https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html)", path)
 	err = os.MkdirAll(filepath.Dir(path), 0700)
 	if err != nil {
 		return err
 	}
 
-	file, err := os.OpenFile(path, os.O_WRONLY|os.O_CREATE|os.O_EXCL, 0600)
+	creds, err := ini.Load(path)
 	if err != nil {
-		return err
+		if os.IsNotExist(err) {
+			creds = ini.Empty()
+			creds.Section("").Comment = "https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html"
+		} else {
+			return err
+		}
 	}
-	defer file.Close()
 
 	profile := os.Getenv("AWS_PROFILE")
 	if profile == "" {
 		profile = "default"
 	}
 
-	return tmpl.Execute(file, map[string]string{
-		"KeyID":     keyID,
-		"Profile":   profile,
-		"SecretKey": secretKey,
-	})
+	creds.Section(profile).Key("aws_access_key_id").SetValue(keyID)
+	creds.Section(profile).Key("aws_secret_access_key").SetValue(secretKey)
+
+	tempPath := path + ".tmp"
```

Comment:
can you add to the commit comment why we are doing this. will be helpful.

Reply:
> can you add to the commit comment why we are doing this.

Done with ed4b050 -> 56e22fa (e167ad3e88b has the updated commit message), which also rebases onto master.